### PR TITLE
cmd/atlas/internal: fix IAM Authentication for RDS MySQL

### PR DESCRIPTION
**Pull request by ronenlu**

### CODE REVIEW

File: cmd/atlas/internal/cmdext/cmdext.go [hunk 1]
```diff
@@ -152,6 +156,24 @@ func AWSRDSToken(ctx *hcl.EvalContext, block *hclsyntax.Block) (cty.Value, error
...
+	}
+	pem, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return cty.NilVal, errorf("reading ca bundle: %v", err)
+	}
+	rootCertPool := x509.NewCertPool()
+	if ok := rootCertPool.AppendCertsFromPEM(pem); !ok {
+		return cty.NilVal, errorf("appending certs from pem: %v", err)
+	}
+	err = mysql.RegisterTLSConfig("rds", &tls.Config{RootCAs: rootCertPool, InsecureSkipVerify: true})
```
Comment (giautm) [inline cmd/atlas/internal/cmdext/cmdext.go:173]: I don't think this fix is right. The user should download it by their own and update its location to the driver.

Comment (a8m) [inline cmd/atlas/internal/cmdext/cmdext.go:173]: I don't like working with drivers directly in packages that are not main (atlas packages don't aware to the underlying driver). I prefer creating a driver for `awsmysql`. e.g., https://github.com/google/go-cloud/blob/master/mysql/awsmysql/awsmysql.go

Comment (rotemtam) [reply cmd/atlas/internal/cmdext/cmdext.go:173]: Is it necessary? I'm guessing its posible to provide the CA path to the driver. 
we shouldn't do the magic of forcing the user to trust some cert. its an advanced use case and the user should manage their own CAs.

Comment (masseelch) [reply cmd/atlas/internal/cmdext/cmdext.go:173]: > Is it necessary? I'm guessing its posible to provide the CA path to the driver. we shouldn't do the magic of forcing the user to trust some cert. its an advanced use case and the user should manage their own CAs.

Yes. It is needed. This is how it works with mysql go sql driver sadly.

**Review by a8m — APPROVED** [review]

**Pull request merged**